src/core/src/energy/global/ewald.rs

```diff
@@ -327,6 +327,21 @@ impl Ewald {
 
 /// k-space part of the summation
 impl Ewald {
+
+    /// Return a vector containing every (ikx, iky, ikz) indexes.
+    ///
+    /// Ideally this would be done using flat_map on a parallel iterator,
+    /// but there is currently no satisfying way to return a FlatMap
+    /// iterator without boxing it.
+    fn get_ik_indexes(&self) -> Vec<(usize, usize, usize)> {
```

Comment:
This function means we are allocating a Vec, and we were not before, is it right?

Why not have a specific struct implementing `Iterator` for this?